### ISSUE TITLE
Building with -extldflags=-static 

### ISSUE THREAD

**Issue by Janso123** [davidbyttow/govips#339]
Hi, 
Is there an option to build it with a static flag? 

That's my Dockerfile and it core-dumps on me or throws linker error
`FROM golang:alpine as builder

RUN apk add musl-dev vips-dev gcc

WORKDIR /src
COPY . .

RUN go build -o /bin/app --ldflags '-linkmode external -extldflags=-static' . `

**Comment by tonimelisma** [comment]
Hey @sonu27 I've never tried it personally.

**Comment by g-mero** [comment]
Same issue.

**Comment by moguchev** [comment]
The same issue

**Comment by omar391** [comment]
Did you try with `musl` libc?

**Comment by moguchev** [comment]
Yes. It's seems that static build is impossible (Or some magic should happen). Without static flag everything is fine.

Here is my final Docker file (build is working on MacOS Sonoma 14.1.1 with M1 and also on Ubuntu):

```
#################################
# STEP 1. Build executable binary
#################################
FROM --platform=linux/amd64 golang:1.21-alpine3.18 AS build

ARG TARGETOS TARGETARCH
ARG BUILDPLATFORM TARGETPLATFORM

RUN echo "I am running on $(uname -m), building for $TARGETPLATFORM, OS=$TARGETOS, ARCH=$TARGETARCH"

# Download gcc (cross-compilers) for CGO
RUN apk update && apk add build-base

# Install required libraries for libvips and deps
RUN apk update && \
    apk add \
    pkgconfig \
	vips-dev

WORKDIR /src

# Download necessary Go modules
COPY go.mod .
COPY go.sum .
RUN CGO_ENABLED=1 GOOS=${TARGETOS} GOARCH=${TARGETARCH} go mod download

# Copy all files
COPY . .

# Build go binary
ENV CC=gcc
ENV GOOS=linux
ENV GOARCH=amd64
ENV CGO_ENABLED=1
RUN go build -v -a -tags vips --ldflags="-s -w -linkmode external" -o /bin/facade -v ./cmd/facade
# CGO_ENABLED=1 GOOS=${TARGETOS} GOARCH=${TARGETARCH} CC=gcc 
# #############################
# # STEP 2. Build a small image
# #############################
FROM --platform=linux/amd64 alpine:3.18.0 AS final

# Add SSL certs
ADD certs/root.crt /etc/ssl/certs/
ADD certs/root.crt ~/.postgresql/
ADD certs/root.crt ~/.opensearch/

# Copy builded libraries from build image
COPY --from=build /bin/facade /facade

RUN apk update && \
    apk add \
    glib \
    libexpat \
    libwebp \
    libwebp-tools \
    libpng \
    libjpeg-turbo \
    vips \
    file 

RUN file /facade 
RUN ldd /facade

EXPOSE 8080
EXPOSE 8082

CMD /facade
```

**Comment by aisnote** [comment]
same issue on windows native building, it worked on libvips-8.10, but can not worked on 8.14.
Anyone has idea?
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lgio-2.0: No such file or directory
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lintl: No such file or directory
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lgmodule-2.0: No such file or directory
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lintl: No such file or directory
C:/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lintl: No such file or directory

**Comment by davidbyttow** [comment]
Closing as won't fix: fully static linking is not feasible with libvips and its GLib/GObject dependencies, which do not ship static libraries in most package managers and have known symbol issues with static linking (e.g. `getpwuid`). The standard approach is to use a dynamic binary with shared libraries in a minimal container image (e.g. Alpine final stage). This is an ecosystem constraint, not something govips can address.
